test/abc/sound/SequencePlayerTest.java

```diff
@@ -1,38 +1,125 @@
 package abc.sound;
 
-import static org.antlr.v4.runtime.CharStreams.fromFileName;
+import javax.sound.midi.InvalidMidiDataException;
+import javax.sound.midi.MidiUnavailableException;
 
-import java.io.IOException;
-
-import org.antlr.v4.runtime.CharStream;
-import org.antlr.v4.runtime.CommonTokenStream;
-import org.antlr.v4.runtime.tree.ParseTree;
-import org.antlr.v4.runtime.tree.ParseTreeWalker;
 import org.junit.Test;
 
-import abc.parser.AbcBaseListener;
-import abc.parser.AbcLexer;
-import abc.parser.AbcParser;
-
 public class SequencePlayerTest {
 
 	// TODO: warmup #2
 
 	@Test
-	public void test() {
-		String fileName = "sample_abc/piece1.abc";
+	public void piece1() {
+		try {
+			SequencePlayer player = new SequencePlayer(140, 12);
+
+			player.addNote(new Pitch('C').toMidiNote(), 0, 12);
+			player.addNote(new Pitch('C').toMidiNote(), 12, 12);
+			player.addNote(new Pitch('C').toMidiNote(), 24, 9);
+			player.addNote(new Pitch('D').toMidiNote(), 33, 3);
+			player.addNote(new Pitch('E').toMidiNote(), 36, 12);
+
+			player.addNote(new Pitch('E').toMidiNote(), 48, 9);
+			player.addNote(new Pitch('D').toMidiNote(), 57, 3);
+			player.addNote(new Pitch('E').toMidiNote(), 60, 9);
+			player.addNote(new Pitch('F').toMidiNote(), 69, 3);
+			player.addNote(new Pitch('G').toMidiNote(), 72, 24);
+
+			player.addNote(new Pitch('D').transpose(12).toMidiNote(), 96, 4);
```

Comment:
All these tuplet notes should have the same `startTick` right?

Reply:
I don't think so. It can have different startTicks. If it has the same, then it doesn't sound well. You can change the startTick to listen to how the music sounds.

Reply:
So I listened to it and I think these three 'd' notes should be 'c' notes.